Failing tests

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behaviour

The following three tests fail in github actions but not on my computer dev environment, it appears to consistently be these three that fail. 

```
FAILED tests/test_lidarr.py::test_get_tracks - IndexError: list index out of range
FAILED tests/test_readarr.py::test_add_metadata_profile - KeyError: 'name'
FAILED tests/test_readarr.py::test_add_quality_profile - KeyError: 'name'
```

### Expected behaviour

All tests should pass in github actions

### Steps To Reproduce

Run github action "test"

### Pyarr Version

5.0.0

### Python Version

3.9,3.10,3.11

### Example Code

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/noxfile.py
+++ b/noxfile.py
@@ -17,7 +17,19 @@ def format(session: Session) -> None:
 def coverage(session: Session) -> None:
     """Check tests cover all code"""
     session.run("poetry", "install", external=True)
-    session.run("pytest", "tests", "--cov=pyarr", "--cov-report", "term-missing", "-vv")
+    session.run(
+        "pytest",
+        "tests",
+        "--showlocals",
+        "--reruns",
+        "3",
+        "--reruns-delay",
+        "5",
+        "--cov=pyarr",
+        "--cov-report",
+        "term-missing",
+        "-vv",
+    )
 
 
 @nox.session(reuse_venv=True)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -50,6 +50,7 @@ autoflake = "^1.7.8"
 ipykernel = "^6.22.0"
 sphinx-toolbox = "^3.4.0"
 enum-tools = "^0.9.0.post1"
+pytest-rerunfailures = "^11.1.2"
 
 [tool.black]
 line-length = 88

